src/pool/mod.rs

```diff
@@ -154,23 +164,32 @@ impl<T: Poolable, K: Key> Drop for Pooled<K, T> {
                 return;
             }
 
+            // Add the connection back to the queue directly if the weak reference is still alive.
+            if let Some(weak_queue) = &self.queue {
+                if let Some(queue) = weak_queue.upgrade() {
+                    let idle = Idle::new(value);
+                    queue.borrow_mut().push_back(idle);
+                    return;
```

Comment:
Have we considered if the max_idle is reached here?

Reply:
Fixed this in the latest commit